### PR TITLE
doc: shields: Update shields doc with missing features

### PR DESCRIPTION
**Pull request by erwango**
Shields subsystem provides several undocumented features (as mentioned in #19086)
Update doc.

Change is split in 2 commits as one part is valid for 2.0 and second part is only available in later versions.

### CODE REVIEW

**Review by jfischer-no — APPROVED** [review]

**Review by mike-scott — APPROVED** [review]
Thank you @erwango for putting this up.  LGTM.

**Review by avisconti — APPROVED** [review]
LGTM

File: doc/guides/porting/shields.rst [hunk 1]
```diff
@@ -26,6 +27,11 @@ These files provides shield configuration as follows:
   format that is merged with the board's device tree information before
   compilation.
 
+* **<shield>.conf**: This file defines values for Kconfig symbols that are
+  required for default shield configuration. Please note that, to ease usage
```
Comment (dbkinder) [inline doc/guides/porting/shields.rst]: For clarity, how about changing this last sentence to something like this:
```
To ease use with applications, the default shield
configurations here should be consistent with those
in the :ref:`default_board_configuration`.
```

File: doc/guides/porting/shields.rst [hunk 2]
```diff
@@ -59,6 +65,20 @@ introduced overriding node element:
 
         arduino_i2c:i2c1{};
 
+Board specific shield configuration
+-----------------------------------
+
+It may happen that some modifications are needed to fit a shield to a particular
```
Comment (dbkinder) [inline doc/guides/porting/shields.rst]: How about instead:
```
If modifications are needed
to fit a shield to a particular board, you
can override a shield description for
a specific board by adding board-overriding
files to a shield, as follows:
```

File: doc/guides/porting/shields.rst [hunk 3]
```diff
@@ -77,3 +97,40 @@ Alternatively, it could be set by default in a project's CMakeLists.txt:
...
+
+.. code-block:: none
+
+   boards/shields/<shield>
+   ├── <shield_v1>.overlay
+   ├── <shield_v1>.conf
+   ├── <shield_v2>.overlay
+   └── <shield_v2>.conf
+
+In that case shield particluar revision should be activated using the revision
```
Comment (dbkinder) [inline doc/guides/porting/shields.rst]: How about (for clarity and fixing a misspelling)
```
In this case, a shield-particular revision name can be used:
```

File: doc/guides/porting/shields.rst [hunk 4]
```diff
@@ -77,3 +97,40 @@ Alternatively, it could be set by default in a project's CMakeLists.txt:
...
+name:
+
+  .. zephyr-app-commands::
+     :zephyr-app: your_app
+     :shield: shield_v2
+     :goals: build
+
+Again, it is also possible to provide a board specific configuration to a
```
Comment (dbkinder) [inline doc/guides/porting/shields.rst]: Change to:
```
You can also provide a board-specific configuration
to a specific shield revision:
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
some recommended clarity edits

**Comment by erwango** [comment]
@dbkinder, updated thanks for review

**Comment by erwango** [comment]
@dbkinder, can you review again? Thanks!

File: doc/guides/porting/shields.rst [hunk 5]
```diff
@@ -26,6 +27,11 @@ These files provides shield configuration as follows:
   format that is merged with the board's device tree information before
   compilation.
 
+* **<shield>.conf**: This file defines values for Kconfig symbols that are
+  required for default shield configuration. Please note that, to ease use with
```
Comment (dbkinder) [inline doc/guides/porting/shields.rst]: While it's polite, we don't need "please" in our documentation.  Delete "Please note that, " and start the sentence with "To ease use ..."

**Review by dbkinder — CHANGES_REQUESTED** [review]
One little tweak, otherwise, LTGM.

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

**Pull request merged**